electrum/interface.py

```diff
@@ -399,7 +399,7 @@ def __init__(self, *, network: 'Network', server: ServerAddr, proxy: Optional[di
         self.taskgroup = OldTaskGroup()
 
         async def spawn_task():
-            task = await self.network.taskgroup.spawn(self.run())
+            task = await self.network.taskgroup.spawn(await self.run())
```

Comment:
`await self.run()` returns None, or at least it would, except it is effectively an infinite-loop which can only exit via exceptions.
`await self.network.taskgroup.spawn(None)` does not make sense and would error, but it never gets to execute.

what you effectively have here between lines 401-404 could then just be replaced with
`asyncio.run_coroutine_threadsafe(self.run(), self.network.asyncio_loop)`

With your change, the interfaces run directly via `run_coroutine_threadsafe` and bypass `network.taskgroup`, which means that they are not closed cleanly if the network is restarted (e.g. due to proxy changes) or during application shutdown.